NASA-PDS/pds-doi-service
# Enhanced validation with XML Schema and regression test suite

Pull request to close issues #41 and #56 .

**Summary***
The changes are to support two issues:

https://github.com/NASA-PDS/pds-doi-service/issues/41 Develop simple regression test suite for deployment
https://github.com/NASA-PDS/pds-doi-service/issues/56 validate the submitted OSTI record against a schema

**Test Data and/or Report**
One of the following should be included here:
* Reference to regression test included in code (preferred wherever reasonable)
   The feature files for the 'behave' testing includes tables with input needed for the 'behave' testing.  The .zip file (ATMOS_mpf_irtf_Bundles_20200414.zip) will need to be unzip and placed in the default run directory, with the top level directory aaDOI_production_submitted_labels.
* Attach test data here + outputs of tests

How to run the unit tests:

        python3 pds_doi_core/actions/test/draft_test.py
        python3 pds_doi_core/actions/test/list_test.py
        python3 pds_doi_core/actions/test/check_test.py
        python3 pds_doi_core/actions/test/reserve_test.py
        python3 pds_doi_core/actions/test/release_test.py
        python3 pds_doi_core/util/test/doi_validator_test.py

How to run the behave test (assume the executable has been installed in user's directory and the behave command is in the path)

        behave --no-capture --no-logcapture >& temp_file.txt; tail -5 temp_file.txt

If everything is OK, you should see something like this at the end of temp_file.txt

2 features passed, 0 failed, 0 skipped
11 scenarios passed, 0 failed, 0 skipped
36 steps passed, 0 failed, 0 skipped, 0 undefined
Took 0m2.230s

If not, inspect the output file for where the error may have occurred.
